docs/resiliency/index.asciidoc

```diff
@@ -156,24 +159,60 @@ Almost all files in Elasticsearch now have checksums which are validated before
 
-* Introduce shard level locks to prevent concurrent shard modifications {GIT}8436[#8436]. (STATUS: DONE, Fixed in v1.5.0)
-* Delete shard contents under a lock {GIT}9083[#9083]. (STATUS: DONE, Fixed in v1.5.0)
-* Delete shard under a lock {GIT}8579[#8579]. (STATUS: DONE, Fixed in v1.5.0)
-* Refactor RecoveryTarget state management {GIT}8092[#8092]. (STATUS: DONE, Fixed in v1.5.0)
-* Cancelling a recovery may leave temporary files behind {GIT}7893[#7893]. (STATUS: DONE, Fixed in v1.5.0)
-* Quick cluster state processing can result in both shard copies being deleted {GIT}9503[#9503]. (STATUS: DONE, Fixed in v1.5.0)
-* Rapid creation and deletion of an index can cause reuse of old index metadata {GIT}9489[#9489]. (STATUS: DONE, Fixed in v1.5.0)
-* Flush immediately after the last concurrent recovery finishes to clear out the translog before a new recovery starts {GIT}9439[#9439]. (STATUS: DONE, Fixed in v1.5.0)
+[float]
+=== Document guarantees and handling of failure (STATUS: ONGOING)
+
+This status page is a start, but we can do a better job of explicitly documenting the processes at work in Elasticsearch, and what happens in the case of each type of failure. The plan is to have a test case that validates each behavior under simulated conditions. Every test will document the expected results, the associated test code and an explicit PASS or FAIL status for each simulated case.
+
+
+[float]
+=== Take filter cache key size into account (STATUS: ONGOING)
+
+Commonly used filters are cached in Elasticsearch. That cache is limited in size (10% of node's memory by default) and is being evicted based on a least recently used policy. The amount of memory used by the cache depends on two primary components - the values it stores and the keys associated with them. Calculating the memory footprint of the values is easy enough but the keys accounting is trickier to achieve as they are, by default, raw Lucene objects. This is largely not a problem as the keys are dominated by the values. However, recent optimizations in Lucene have changed the balance causing the filter cache to grow beyond it's size.
+
+While we are working on a longer term solution ({GIT}9176[#9176]), we introduced a minimum weight of 1k for each cache entry. This puts an effective limit on the number of entries in the cache. See {GIT}8304[#8304] (STATUS: DONE, fixed in v1.4.0)
+
+== Completed
 
 [float]
-=== Prevent setting minimum_master_nodes to more than the current node count (STATUS: ONGOING)
+=== Ensure cluster state ID is incremental (STATUS: DONE, v1.5.1)
+
+It is possible that the current state ID can be reset or even go backwards
+when nodes or the cluster are restarted, or nodes with legacy data join
```

Comment:
We never full researched how this could have happened. As far as I can tell, this can happen on a full cluster restart (or when all copies of the shard are gone) which happens while the nodes are writing these state file. On top of that the new shards have to be assigned before all shard copies are present back in the cluster. I would reword this sentence to something like ".. go backwards in very extreme cases during full cluster restart".